components/form-builder/store/useTemplateStore.tsx

```diff
@@ -432,6 +435,15 @@ export const useTemplateStore = <T,>(
   return useStore(store, selector, equalityFn);
 };
 
+export const useSubscibeToTemplateStore = <T,>(
+  selector: (state: TemplateStoreState) => T,
+  listener: (selectedState: T, previousSelectedState: T) => void
+) => {
+  const store = useContext(TemplateStoreContext);
+  if (!store) throw new Error("Missing Template Store Provider in tree");
+  useEffect(() => store.subscribe(selector, listener), [store, selector, listener]);
```

Comment:
Is the `store` dep here the store from the const (line 442)?

Reply:
It is.  I included it for the time being because of the `exhaustive-deps` hooks rule but haven't had the time to test yet whether or not state changes to the store fire the UseEffect.